Tests/ResultTests.swift

```diff
@@ -277,4 +280,194 @@ class ResultTestCase: BaseTestCase {
+        // Then
+        if let error = mappedResult.error {
+            XCTAssertTrue(error is ThrownError)
+        } else {
+            XCTFail("mapError should capture thrown error value")
+        }
+    }
+    
+    // MARK: with/if Tests
+    
+    func testWithValueExecutesWhenSuccess() {
+        // Given
+        let result: Result<String> = .success("success")
+        var string = "failure"
+        
+        // When
+        result.withValue { string = $0 }
+        
+        // Then
+        XCTAssertEqual(string, "success")
+    }
+    
+    func testWithValueDoesNotExecutesWhenFailure() {
+        // Given
+        struct ResultError: Error {}
+        let result: Result<String> = .failure(ResultError())
+        var string = "failure"
+        
+        // When
+        result.withValue { string = $0 }
+        
+        // Then
+        XCTAssertEqual(string, "failure")
+    }
+    
+    func testWithErrorExecutesWhenFailure() {
+        // Given
+        struct ResultError: Error {}
+        let result: Result<String> = .failure(ResultError())
+        var string = "success"
+        
+        // When
+        result.withError { string = "\(type(of: $0))" }
+        
+        // Then
+        XCTAssertEqual(string, "(ResultError #1)")
+    }
+    
+    func testWithErrorDoesNotExecuteWhenSuccess() {
+        // Given
+        let result: Result<String> = .success("success")
+        var string = "success"
+        
+        // When
+        result.withError { string = "\(type(of: $0))" }
+        
+        // Then
+        XCTAssertEqual(string, "success")
+    }
+    
+    func testIfSuccessExecutesWhenSuccess() {
+        // Given
+        let result: Result<String> = .success("success")
+        var string = "failure"
+        
+        // When
+        result.ifSuccess { string = "success" }
+        
+        // Then
+        XCTAssertEqual(string, "success")
+    }
+    
+    func testIfSuccessDoesNotExecutesWhenFailure() {
+        // Given
+        struct ResultError: Error {}
+        let result: Result<String> = .failure(ResultError())
+        var string = "failure"
+        
+        // When
+        result.ifSuccess { string = "success" }
+        
+        // Then
+        XCTAssertEqual(string, "failure")
+    }
+    
+    func testIfFailureExecutesWhenFailure() {
+        // Given
+        struct ResultError: Error {}
+        let result: Result<String> = .failure(ResultError())
+        var string = "success"
+        
+        // When
+        result.ifFailure { string = "failure" }
+        
+        // Then
+        XCTAssertEqual(string, "failure")
+    }
+    
+    func testIfFailureDoesNotExecuteWhenSuccess() {
+        // Given
+        let result: Result<String> = .success("success")
+        var string = "success"
+        
+        // When
+        result.ifFailure { string = "failure" }
+        
+        // Then
+        XCTAssertEqual(string, "success")
+    }
+    
```

Comment:
Addressed in [9bf29](https://github.com/Alamofire/Alamofire/pull/2135/commits/131354b48086ea94f3b3d9303347d2f64079bf29).